app/src/openxr/cpp/OpenXRLayers.h

```diff
@@ -185,8 +185,13 @@ class OpenXRLayerBase : public OpenXRLayer {
     info.height = height;
     bool shouldZeroInitialize = aSurfaceType == VRLayerSurface::SurfaceType::AndroidSurface;
 #if defined(PICOXR)
-    // Circumvent a bug in the pico OpenXR runtime.
-    shouldZeroInitialize = false;
+    // Circumvent a bug in the pico OpenXR runtime in versions below 5.4.0.
+    char buildId[128] = {0};
+    __system_property_get("ro.build.id", buildId);
```

Comment:
What about converting this into a function in the new file with utilities that I mentioned in some other comment? Note that this is not Pico specific. I've just found out that you could actually do
```
__system_property_get(ANDROID_OS_BUILD_ID, buildId)
```
as it's defined in `<sys/system_properties.h>`

Reply:
Done.

Reply:
It doesn't seem that `ANDROID_OS_BUILD_ID` is defined in the public headers. People found out its value by looking at the source code or by using something like `adb shell getprop`.

I have moved the code to get that property to a helper function.